src/util/gifRecorder.ts

```diff
@@ -0,0 +1,70 @@
+import { Recorder } from './recorder';
+import gifEncoder = require('gif-encoder-2-browser');
+import { createCanvas } from 'canvas';
+
+interface GifRecordingContext {
+  interval: ReturnType<typeof setInterval>;
+  frames: ImageData[];
+  height: number;
+  width: number;
+  frameDuration: number;
+  onDataReady: (data: Blob) => void;
+}
+
+export class GifRecorder extends Recorder<Blob, GifRecordingContext> {
+  protected readonly format = 'GIF';
+  constructor(private readonly canvas: HTMLCanvasElement) {
+    super();
+  }
+
+  protected startRecordingImplementation(onDataReady: (data: Blob) => void): GifRecordingContext {
+    const canvas = this.canvas;
+    const height = canvas.height;
+    const width = canvas.width;
+    const frameDuration = 80;
+    const frames: ImageData[] = [];
+
+    const interval = setInterval(() => {
+      const context = this.canvas?.getContext('2d');
+      if (context) {
+        frames.push(context.getImageData(0, 0, width, height));
+      }
+    }, frameDuration);
+
+    return {
+      interval: interval,
+      frames: frames,
+      height: height,
+      width: width,
+      frameDuration: frameDuration,
+      onDataReady: onDataReady,
```

Comment:
```suggestion
      interval,
      frames,
      height,
      width,
      frameDuration,
      onDataReady,
```

You could use this shortcut.